src/instance.rs

```diff
@@ -45,9 +45,8 @@ pub unsafe trait PyNativeType: Sized {
 #[repr(transparent)]
 pub struct Py<T>(NonNull<ffi::PyObject>, PhantomData<T>);
 
-unsafe impl<T> Send for Py<T> {}
-
-unsafe impl<T> Sync for Py<T> {}
+unsafe impl<T: Send> Send for Py<T> {}
+unsafe impl<T: Sync> Sync for Py<T> {}
```

Comment:
This isn't correct. `Py<PyAny>` should be both `Send` and `Sync`, but `PyAny` is neither.